nexuse2e-webapp/src/main/frontend/src/app/filter-panel/filter-panel.component.ts

```diff
@@ -18,17 +19,38 @@ export class FilterPanelComponent implements OnInit {
 
   constructor(
     public screenSizeService: ScreensizeService,
-    public sessionService: SessionService
+    public sessionService: SessionService,
+    private route: ActivatedRoute
   ) {}
 
-  ngOnInit(): void {
-    this.activeFilters = this.sessionService.getActiveFilters(this.itemType);
+  ngOnInit() {
+    const filtersFromRoute = this.extractFiltersFromRouteParams();
+    this.activeFilters = Object.keys(filtersFromRoute).length
+      ? filtersFromRoute
+      : this.sessionService.getActiveFilters(this.itemType);
     if (Object.keys(this.activeFilters).length === 0) {
       this.setDefaultValues();
     }
     this.applyFilters();
   }
 
+  extractFiltersFromRouteParams(): ActiveFilterList {
+    const params = Object.assign({}, this.route.snapshot.queryParams);
+    const startEndDateValue = this.route.snapshot.queryParamMap.get(
+      "startEndDateRange"
+    );
+    if (startEndDateValue) {
+      const startEndDateRange = JSON.parse(startEndDateValue);
+      startEndDateRange.startDate = new Date(startEndDateRange.startDate);
+      startEndDateRange.endDate = new Date(startEndDateRange.endDate);
+      const convertedParams = {
+        startEndDateRange: startEndDateRange,
```

Comment:
Since property name and variable have the same name, we can shorten this
```suggestion
        startEndDateRange,
```

Reply:
Perfect, thanks!